src/rdkafka_admin.c

```diff
@@ -9126,3 +9127,275 @@ void rd_kafka_DescribeCluster(rd_kafka_t *rk,
+                                    reply,
+                                    "Broker returned partition %s [%" PRId32
+                                    "] that was not "
+                                    "included in the original request",
+                                    topic, partition);
+                        }
+
+                        if (rd_list_elem(&partitions_arr, orig_pos) != NULL) {
+                                rd_kafka_buf_parse_fail(
+                                    reply,
+                                    "Broker returned partition %s [%" PRId32
+                                    "] multiple times",
+                                    topic, partition);
+                        }
+
+                        rd_list_set(&partitions_arr, orig_pos,
+                                    partition_result);
+                }
+                rd_kafka_buf_skip_tags(reply);
+        }
+
+        rd_kafka_buf_skip_tags(reply);
+
+        result = rd_kafka_ElectLeadersResult_new(&partitions_arr);
+
+        rko_result = rd_kafka_admin_result_new(rko_req);
+
+        rd_list_init(&rko_result->rko_u.admin_result.results, 1,
+                     rd_kafka_ElectLeadersResult_free);
+
+        rd_list_add(&rko_result->rko_u.admin_result.results, result);
+
+        *rko_resultp = rko_result;
+
+        rd_list_destroy(&partitions_arr);
+
+        return RD_KAFKA_RESP_ERR_NO_ERROR;
+err_parse:
+
+        rd_list_destroy(&partitions_arr);
+
+        if (rko_result)
+                rd_kafka_op_destroy(rko_result);
+
+        rd_snprintf(errstr, errstr_size,
+                    "ElectLeaders response protocol parse failure: %s",
+                    rd_kafka_err2str(reply->rkbuf_err));
+
+        return reply->rkbuf_err;
+}
+
+void rd_kafka_ElectLeaders(rd_kafka_t *rk,
+                           rd_kafka_ElectLeaders_t *elect_leaders,
+                           const rd_kafka_AdminOptions_t *options,
+                           rd_kafka_queue_t *rkqu) {
+        rd_kafka_op_t *rko;
+        rd_kafka_topic_partition_list_t *copied_partitions;
+
+        static const struct rd_kafka_admin_worker_cbs cbs = {
+            rd_kafka_ElectLeadersRequest,
+            rd_kafka_ElectLeadersResponse_parse,
+        };
+
+        rd_assert(rkqu);
+
+        rko = rd_kafka_admin_request_op_new(rk, RD_KAFKA_OP_ELECTLEADERS,
+                                            RD_KAFKA_EVENT_ELECTLEADERS_RESULT,
+                                            &cbs, options, rkqu->rkqu_q);
+
+        /* Non empty topic_partition_list should be present */
+        if (elect_leaders->partitions->cnt == 0) {
+                rd_kafka_admin_result_fail(rko, RD_KAFKA_RESP_ERR__INVALID_ARG,
+                                           "No partitions specified");
+                rd_kafka_admin_common_worker_destroy(rk, rko,
+                                                     rd_true /*destroy*/);
+                return;
+        }
```

Comment:
I was going through the Java implementation and figured out that null list is also allowed which will trigger leader election for all the topic partitions. We need to change the code to support this. There maybe multiple places we need to do this change.

https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/admin/Admin.java#L1040-L1041

Topic Partition is nullable - https://github.com/apache/kafka/blob/trunk/clients/src/main/resources/common/message/ElectLeadersRequest.json#L29